src/foam/lib/json/Outputter.java

```diff
@@ -96,17 +97,26 @@ protected void outputNull() {
   }
 
   public void outputString(String s) {
+    boolean hasNewLine = s.indexOf('\n') >= 0;
+    if ( multiLineOutput_ && hasNewLine ) writer_.append("\r\n  ");
     writer_.append("\"");
+    if ( multiLineOutput_ && hasNewLine ) writer_.append("\"\"\r\n    ");
     writer_.append(escape(s));
+    if ( multiLineOutput_ && hasNewLine ) writer_.append("\r\n  \"\"");
     writer_.append("\"");
   }
 
   public String escape(String s) {
-    return s
-      .replace("\\", "\\\\")
+    boolean hasNewLine = s.indexOf('\n') >= 0;
```

Comment:
I would rather have two escape methods for single vs multiline strings, otherwise we're searching the whole string for newlines _again_ every time.